packages/main/src/components/AnalyticalTable/hooks/useDynamicColumnWidths.ts

```diff
@@ -0,0 +1,155 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { DEFAULT_COLUMN_WIDTH } from '../defaults/Column';
+import { TableScaleWidthMode } from '@ui5/webcomponents-react/lib/TableScaleWidthMode';
+
+const ROW_SAMPLE_SIZE = 20;
+const DEFAULT_HEADER_NUM_CHAR = 10;
+const MAX_WIDTH = 700;
+
+// a function, which approximates header px sizes given a character length
+const approximateHeaderPxFromCharLength = (charLength) =>
+  charLength < 15 ? Math.sqrt(charLength * 1500) : 8 * charLength;
+const approximateContentPxFromCharLength = (charLength) => 8 * charLength;
+
+export const useDynamicColumnWidths = (scaleWidthMode, columns, rows, totalWidth, hiddenColumns, setColumnWidth) => {
```

Comment:
can we run this as a tableHook? just the our useXYZStyle hooks? then we would have access to all table data like columns, row, width, hidden columns, etc...